### PR TITLE
NEO-51282: Add support for xtk:jobInterface methods

### PR DESCRIPTION
**Pull request by mkiki**
## Description

The xtk:jobInterface is not currently supported by the SDK. The reason is that job entities do not implement xtk:jobInterface (they implement xtk:job) and job interface methods are therefore not visible to the SDK.

The SDK now provides the client.jobInterface function which returns an object which can be used to submit jobs, retreive their status, progress, logs, etc. See the documentation of the SDK for more details.

## Motivation and Context

Long running jobs such as nms:delivery#Prepare must be run as jobs

## How Has This Been Tested?

* New unit test for all new code
* Actually tested on local instance

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/xtkJob.html [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  
+<p>
+  Simple jobs are non-static methods with no parameters, such as the nms:delivery#Prepare method which is used to prepare a delivery. 
+  Such jobs can be executed synchronously (xtk:jobInterface#Execute) or asynchronously (xtk:jobInterface#Submit). Both methods return
+  a job id which can be used to retreive (poll) more about the job.
+<p>
+    
+<p>
+  First, we need to retreive a delivery, which is the object upon which to run the Prepare method, but also implements xtk:job interface. 
+  The Prepare method requires a complete delivery object, so we use SelectAll
```
Comment (mukulgupta21) [inline docs/xtkJob.html:23]: For my knowledge, does 'Prepare' method really require the complete delivery object? What happens if we only provide the keys?

File: src/client.js [hunk 2]
```diff
@@ -1820,6 +1827,15 @@ class Client {
         const result = this._toRepresentation(doc);
         return result;
     }
+
+    /**
+     * Creates a Job object which can be used to submit jobs, retreive status, logs and progress, etc.
```
Comment (mukulgupta21) [inline src/client.js:1832]:
```suggestion
     * Creates a Job object which can be used to submit jobs, retrieve status, logs and progress, etc.
```

**Review by mukulgupta21 — APPROVED** [review]
👍 . Really like this API. Makes interfacing with jobs so much easier.

File: docs/xtkJob.html (hunk 1 above)
Comment (mkiki) [reply docs/xtkJob.html:23]: Yes, it does. The delivery object is passed directly to the JavaScript nms_delivery_Prepare(elDelivery) function.
Also, the job mechanism itself depends on the folder-id attribute and a couple of others.

**Pull request merged**